### PR TITLE
feat: introduces BLS signature verification precompile

### PR DESCRIPTION
**Pull request by ckartik**
# Introduction of BLS Signature Verification Precompile

This PR introduces a new precompiled contract for **BLS signature verification** using the **BLS12-381 curve**. The `bls12381SignatureVerification` struct enables Ethereum smart contracts to efficiently verify BLS signatures (e.g validate a message was signed by a target public key). This will be useful in the context of ensuring that builders who register have access to the BLS key they sign up with.

### Details for usage 
- **Gas Cost**: Is currently set to **150000 wei.**
- **Input Format**:  The input is a concatenation of three components:

```plaintext
+---------------------+---------------------+-----------------------+
|     Public Key      |       Message       |       Signature       |
|     (48 bytes)      |      (32 bytes)     |       (96 bytes)      |
|      G1 Point       | Hash of the Message |       G2 Point        |
+---------------------+---------------------+-----------------------+
```
Note: We assume the public key is a point on the G1 curve. 

## Example Usage: Solidity Contract

The following example demonstrates how to interact with the BLS signature verification precompile via a smartcontract:

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;

contract BLSVerifier {
    // Precompile address
    address constant BLS_VERIFY = address(0xf0);

    /**
     * @dev Verifies a BLS signature using the precompile
     * @param pubKey The public key (48 bytes G1 point)
     * @param message The message hash (32 bytes)
     * @param signature The signature (96 bytes G2 point)
     * @return success True if verification succeeded
     */
    function verifySignature(
        bytes calldata pubKey,
        bytes32 message,
        bytes calldata signature
    ) public view returns (bool) {
        // Input validation
        require(pubKey.length == 48, "Public key must be 48 bytes");
        require(signature.length == 96, "Signature must be 96 bytes");

        // Concatenate inputs in required format:
        // [pubkey (48 bytes) | message (32 bytes) | signature (96 bytes)]
        bytes memory input = bytes.concat(
            pubKey,
            message,
            signature
        );

        // Call precompile
        (bool success, bytes memory result) = BLS_VERIFY.staticcall(input);
        
        // Check if call was successful
        if (!success) {
            return false;
        }

        // If we got a result back and it's not empty, verification succeeded
        return result.length > 0;
    }
}
```

The video demo can be seen [here](https://www.loom.com/share/d3f957f6bd8c48139d5263fe97aef252?sid=98fef86b-1a49-4d84-a9b3-6f64e19c6cd1)

### CODE REVIEW

**Review by Mikelle — APPROVED** [review]

**Comment by chrmatt** [comment]
If the example use case is just in the PR description, it will get lost when merging. Maybe worth putting this somehwere in a readme?

File: core/vm/contracts.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ import (
 	"fmt"
 	"math/big"
 
+	"github.com/cloudflare/circl/sign/bls"
```
Comment (chrmatt) [inline core/vm/contracts.go:26]: This is not available for me. Do they implement actual BLS verifcation on top of precompiles? Is this documented somewhere?

Comment (ckartik) [reply core/vm/contracts.go:26]: The precompile is implemented by us. They simply implement BLS signatures.

Comment (ckartik) [reply core/vm/contracts.go:26]: take a look here: https://pkg.go.dev/github.com/cloudflare/circl/sign/bls

**Pull request merged**